pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -1513,20 +1523,25 @@ by more specific methods, like assertSame, assertNotSame.
                 <value>
 <![CDATA[
 //PrimaryExpression[
-    PrimaryPrefix/Name
```

Comment:
These changes now create a new false-negative, see https://chunk.io/pmd/f0f07b25b93b453e993fa6b0d71b44e6/diff/spring-framework/index.html#A1 
... not sure why...

Reply:
Might be because of typeresolution:

Before: `ancestor::ClassOrInterfaceDeclaration[//ClassOrInterfaceType[pmd-java:typeIs('junit.framework.TestCase')]]`
After: `ancestor::ClassOrInterfaceDeclaration[pmd-java:typeIs('junit.framework.TestCase')]`

Maybe we should keep it like this?

`ancestor::ClassOrInterfaceDeclaration[ExtendsList/ClassOrInterfaceType[pmd-java:typeIs('junit.framework.TestCase')]]`

It would only affect if typeresolution doesn't have access to the whole auxclasspath.

Hm... that's not actually the issue here: Before, we found the `@Test` annotation on the main class in this file and assumed, also the other classes are tests. This is probably not correct, since ExposedInvocationTestBean or TestBean is not a test case....

Reply:
The thing is this rule is not valid only in test classes. It applies to any usage of `Assert.assertTrue`. Type resolution doesn't give access to which method is invoked, so we try to see if we're in a test class by finding an `@Test` to see if a call to a method named `assertTrue` is really a junit call, instead of a user defined function. This is just a heuristic. Maybe we should change it to check if there are imports for `org.junit`

Reply:
Most rules in this area check, that they are analyzing a TestCase/`@Test` annotated method. But there is one exception: [UseAssertTrueInsteadOfAssertEquals](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#useasserttrueinsteadofassertequals) does not do this check like the others...

Maybe it makes sense to don't restrict the rules (like [UseAssertSameInsteadOfAssertTrue](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#useassertsameinsteadofasserttrue)) to test classes only? That would definitively solve the performance problems with these rules ...